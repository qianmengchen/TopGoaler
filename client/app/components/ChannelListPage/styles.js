import { StyleSheet } from 'react-native';

const board = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    flex: 1,
    alignItems: 'stretch'
  },
  actions: {
    flex: 0.1,
    flexDirection: 'row'
  },
  channelList: {
    marginHorizontal: 8,
    flex: 0.9,
    flexDirection: 'column'
  }
});

const actions = StyleSheet.create({
  searchBar: {
    borderTopWidth: 0,
    borderBottomWidth: 0,
    flex: 0.9,
    backgroundColor: 'transparent',
    justifyContent: 'center',
    alignItems: 'flex-start',
    paddingBottom: 20
  },
  button: {
    flex: 0.1,
    alignItems: 'flex-start',
    justifyContent: 'center',
    paddingBottom: 10
  }
});

export { board, actions };
