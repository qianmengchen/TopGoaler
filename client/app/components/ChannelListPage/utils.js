export const goToChannel = (subscribed, navigate) => channel_id => {
  if (subscribed) navigate('ChannelMemberView', { channel_id });
  else navigate('ChannelPublicView', { channel_id });
};
