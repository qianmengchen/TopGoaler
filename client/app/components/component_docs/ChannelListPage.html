<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: ChannelListPage</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: ChannelListPage</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>ChannelListPage<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="ChannelListPage"><span class="type-signature"></span>new ChannelListPage<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>userId</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last">A unique number identifying the user.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>channels</code></td>
            

            <td class="type">
            
                
<span class="param-type">Object</span>


            
            </td>

            

            

            <td class="description last">An object with the key as the id of the channel and the value as the channel.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>subscribed_channels-</code></td>
            

            <td class="type">
            
                
<span class="param-type">set</span>


            
            </td>

            

            

            <td class="description last">A set that describes those channels that the user is subscribed to.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>addChannel</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last">A function that allows the user to create a new channel.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>subscribe</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last">A function that allows the user to susbcribe to a particular channel</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="ChannelListPage.js.html">ChannelListPage.js</a>, <a href="ChannelListPage.js.html#line39">line 39</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="ChannelListPage.html">ChannelListPage</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sun Dec 02 2018 21:01:21 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>