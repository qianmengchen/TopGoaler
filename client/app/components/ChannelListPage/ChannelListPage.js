import React, { Component } from 'react';
import { ScrollView, View, Text } from 'react-native';
import { Button, SearchBar, Icon } from 'react-native-elements';
import { board, actions } from './styles';
import { ChannelOverview } from '../ChannelOverview/index';
import { goToChannel } from './utils';

class ChannelListPage extends Component {
  _addChannel() {
    this.props.navigation.navigate('NewChannelPage');
  }
  _subscribe = (channel_id, user_id) => () => {
    this.props.subscribe(parseInt(channel_id), user_id);
  };

  _handleSearch(text) {
    this.setState({ input: text });
  }

  _contains = channel => {
    return channel.title
      ? channel.title.toLowerCase().includes(this.state.input.toLowerCase())
      : false;
  };

  static navigationOptions = {
    headerVisible: false,
    headerStyle: { height: 0 }
  };

  /**
   * This contains searchBar, add icon and a list of channelOverview components.
   * It is the first page that user will be redirected to after signing in.
   * 
   * @class ChannelListPage
   * @property {number} userId - A unique number identifying the user.
   * @property {Object} channels - An object with the key as the id of the channel and the value as the channel.
   * @property {set} subscribed_channels- A set that describes those channels that the user is subscribed to.
   * @property {function} addChannel - A function that allows the user to create a new channel.
   * @property {function} subscribe - A function that allows the user to susbcribe to a particular channel
     
   */
  constructor(props) {
    super(props);
    this.state = {
      input: ''
    };
  }

  render() {
    const { navigation, userId } = this.props;

    return (
      <View style={board.container}>
        <ScrollView>
          <View style={{ marginLeft: 10, marginTop: 40 }}>
            <Text style={{ fontSize: 24, fontFamily: 'Avenir-Black' }}>
              Search
            </Text>
          </View>

          <View style={board.actions}>
            <SearchBar
              containerStyle={actions.searchBar}
              lightTheme={true}
              clearIcon={false}
              onChangeText={this._handleSearch.bind(this)}
              onClear={() => {}}
              placeholder="Search for channels"
              icon={{ type: 'font-awesome', name: 'search' }}
            />
            <Button
              containerStyle={actions.button}
              buttonStyle={{ backgroundColor: 'transparent' }}
              icon={<Icon name="add" size={30} color="black" />}
              title=""
              onPress={this._addChannel.bind(this)}
            />
          </View>

          <View style={{ marginLeft: 12, marginTop: 5, marginBottom: 10 }}>
            <Text style={{ fontSize: 24, fontFamily: 'Avenir-Black' }}>
              List
            </Text>
          </View>

          <View style={board.channelList}>
            {Object.keys(this.props.channels).map(id => {
              const ch = this.props.channels[id];
              const check_subscribed = this.props.subscribed_channels
                ? this.props.subscribed_channels.has(parseInt(id))
                : false;
              let channel_overview;
              if (this._contains(ch)) {
                channel_overview = (
                  <ChannelOverview
                    key={id}
                    channel={ch}
                    is_subscribed={check_subscribed}
                    goToChannel={goToChannel(
                      check_subscribed,
                      navigation.navigate
                    )}
                    subscribe={this._subscribe(id, userId)}
                  />
                );
              }
              return channel_overview;
            })}
          </View>
        </ScrollView>
      </View>
    );
  }
}

export default ChannelListPage;
