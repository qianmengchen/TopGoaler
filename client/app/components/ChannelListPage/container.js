import { connect } from 'react-redux';
import ChannelListPage from './ChannelListPage';
import {
  fetchTasks,
  navigateTo,
  createChannelAsUser,
  subscribeChannel
} from '../../actions';

const mapStateToProps = state => ({
  channels: state.database.channel_creator
});

const mapDispatchToProps = dispatch => ({
  refreshTasks: () => dispatch(fetchTasks()),
  reportNav: () => dispatch(navigateTo('TaskDetail')),
  addChannel: (channel, user) => dispatch(createChannelAsUser(channel, user)),
  subscribe: channel => dispatch(subscribeChannel(channel))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ChannelListPage);
