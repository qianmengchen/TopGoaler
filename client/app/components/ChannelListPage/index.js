import ChannelListPage from './container';

export { ChannelListPage };
