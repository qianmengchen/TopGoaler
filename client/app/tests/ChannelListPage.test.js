import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
import sinon from 'sinon';
import { Button, SearchBar } from 'react-native-elements';
require('isomorphic-fetch');

import ChannelListPage from '../components/ChannelListPage/ChannelListPage';

const middlewares = [thunkMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialTaskState = {
  tasks: ['do frontend', 'do backend', 'do MVP ftw'],
  isFetching: false,
  currentTaskPage: 'ChannelListPage'
};

describe('Testing Channel List Page', () => {
  const channels = { map: jest.fn() };
  const navigation = { navigate: jest.fn() };
  const addChannel = sinon.spy(ChannelListPage.prototype, '_addChannel');

  const wrapper = shallow(
    <ChannelListPage channels={channels} navigation={navigation} />,
    {
      context: { store: mockStore(initialTaskState) }
    }
  );
  const render = wrapper.dive();

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('should have 1 Button to add channel', () => {
    expect(render.find(Button)).toHaveLength(1);
  });

  it('should have 1 SearchBar to filter channels', () => {
    expect(render.find(SearchBar)).toHaveLength(1);
  });

  it('should invoke corresponding methods when add channel button is pressed', () => {
    const addChannelBtn = render.find(Button);
    addChannelBtn.simulate('press');
    expect(addChannel.calledOnce).toBe(true);
  });
});
