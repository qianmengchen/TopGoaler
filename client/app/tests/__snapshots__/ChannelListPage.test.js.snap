// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Channel List Page renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#eee",
        "flex": 1,
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    <ScrollViewMock>
      <Component
        style={
          Object {
            "marginLeft": 10,
            "marginTop": 40,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "Avenir-Black",
              "fontSize": 24,
            }
          }
        >
          Search
        </Text>
      </Component>
      <Component
        style={
          Object {
            "flex": 0.13,
            "flexDirection": "row",
            "marginTop": 5,
          }
        }
      >
        <ForwardRef(Themed.SearchBar)
          clearIcon={false}
          containerStyle={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "transparent",
              "borderBottomWidth": 0,
              "borderTopWidth": 0,
              "flex": 0.9,
              "justifyContent": "center",
              "paddingBottom": 20,
            }
          }
          icon={
            Object {
              "name": "search",
              "type": "font-awesome",
            }
          }
          lightTheme={true}
          onChangeText={[Function]}
          onClear={[Function]}
          placeholder="Search for channels"
        />
        <ForwardRef(Themed.Button)
          buttonStyle={
            Object {
              "backgroundColor": "transparent",
            }
          }
          containerStyle={
            Object {
              "alignItems": "flex-start",
              "flex": 0.1,
              "justifyContent": "center",
              "paddingBottom": 10,
            }
          }
          icon={
            <Themed.Icon
              color="black"
              name="add"
              size={30}
            />
          }
          onPress={[Function]}
          title=""
        />
      </Component>
      <Component
        style={
          Object {
            "marginBottom": 10,
            "marginLeft": 12,
            "marginTop": 5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "Avenir-Black",
              "fontSize": 24,
            }
          }
        >
          List
        </Text>
      </Component>
      <Component
        style={
          Object {
            "flex": 0.9,
            "flexDirection": "column",
            "marginHorizontal": 8,
          }
        }
      />
    </ScrollViewMock>
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ScrollViewMock>
        <Component
          style={
            Object {
              "marginLeft": 10,
              "marginTop": 40,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "Avenir-Black",
                "fontSize": 24,
              }
            }
          >
            Search
          </Text>
        </Component>
        <Component
          style={
            Object {
              "flex": 0.13,
              "flexDirection": "row",
              "marginTop": 5,
            }
          }
        >
          <ForwardRef(Themed.SearchBar)
            clearIcon={false}
            containerStyle={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0,
                "borderTopWidth": 0,
                "flex": 0.9,
                "justifyContent": "center",
                "paddingBottom": 20,
              }
            }
            icon={
              Object {
                "name": "search",
                "type": "font-awesome",
              }
            }
            lightTheme={true}
            onChangeText={[Function]}
            onClear={[Function]}
            placeholder="Search for channels"
          />
          <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "transparent",
              }
            }
            containerStyle={
              Object {
                "alignItems": "flex-start",
                "flex": 0.1,
                "justifyContent": "center",
                "paddingBottom": 10,
              }
            }
            icon={
              <Themed.Icon
                color="black"
                name="add"
                size={30}
              />
            }
            onPress={[Function]}
            title=""
          />
        </Component>
        <Component
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 12,
              "marginTop": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "Avenir-Black",
                "fontSize": 24,
              }
            }
          >
            List
          </Text>
        </Component>
        <Component
          style={
            Object {
              "flex": 0.9,
              "flexDirection": "column",
              "marginHorizontal": 8,
            }
          }
        />
      </ScrollViewMock>,
      "style": Object {
        "alignItems": "stretch",
        "backgroundColor": "#eee",
        "flex": 1,
        "paddingLeft": 8,
        "paddingRight": 8,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "marginLeft": 10,
                "marginTop": 40,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Avenir-Black",
                  "fontSize": 24,
                }
              }
            >
              Search
            </Text>
          </Component>,
          <Component
            style={
              Object {
                "flex": 0.13,
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <ForwardRef(Themed.SearchBar)
              clearIcon={false}
              containerStyle={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0,
                  "borderTopWidth": 0,
                  "flex": 0.9,
                  "justifyContent": "center",
                  "paddingBottom": 20,
                }
              }
              icon={
                Object {
                  "name": "search",
                  "type": "font-awesome",
                }
              }
              lightTheme={true}
              onChangeText={[Function]}
              onClear={[Function]}
              placeholder="Search for channels"
            />
            <ForwardRef(Themed.Button)
              buttonStyle={
                Object {
                  "backgroundColor": "transparent",
                }
              }
              containerStyle={
                Object {
                  "alignItems": "flex-start",
                  "flex": 0.1,
                  "justifyContent": "center",
                  "paddingBottom": 10,
                }
              }
              icon={
                <Themed.Icon
                  color="black"
                  name="add"
                  size={30}
                />
              }
              onPress={[Function]}
              title=""
            />
          </Component>,
          <Component
            style={
              Object {
                "marginBottom": 10,
                "marginLeft": 12,
                "marginTop": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Avenir-Black",
                  "fontSize": 24,
                }
              }
            >
              List
            </Text>
          </Component>,
          <Component
            style={
              Object {
                "flex": 0.9,
                "flexDirection": "column",
                "marginHorizontal": 8,
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Avenir-Black",
                  "fontSize": 24,
                }
              }
            >
              Search
            </Text>,
            "style": Object {
              "marginLeft": 10,
              "marginTop": 40,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Search",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontFamily": "Avenir-Black",
                "fontSize": 24,
              },
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ForwardRef(Themed.SearchBar)
                clearIcon={false}
                containerStyle={
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": 0,
                    "borderTopWidth": 0,
                    "flex": 0.9,
                    "justifyContent": "center",
                    "paddingBottom": 20,
                  }
                }
                icon={
                  Object {
                    "name": "search",
                    "type": "font-awesome",
                  }
                }
                lightTheme={true}
                onChangeText={[Function]}
                onClear={[Function]}
                placeholder="Search for channels"
              />,
              <ForwardRef(Themed.Button)
                buttonStyle={
                  Object {
                    "backgroundColor": "transparent",
                  }
                }
                containerStyle={
                  Object {
                    "alignItems": "flex-start",
                    "flex": 0.1,
                    "justifyContent": "center",
                    "paddingBottom": 10,
                  }
                }
                icon={
                  <Themed.Icon
                    color="black"
                    name="add"
                    size={30}
                  />
                }
                onPress={[Function]}
                title=""
              />,
            ],
            "style": Object {
              "flex": 0.13,
              "flexDirection": "row",
              "marginTop": 5,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "clearIcon": false,
                "containerStyle": Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0,
                  "borderTopWidth": 0,
                  "flex": 0.9,
                  "justifyContent": "center",
                  "paddingBottom": 20,
                },
                "icon": Object {
                  "name": "search",
                  "type": "font-awesome",
                },
                "lightTheme": true,
                "onChangeText": [Function],
                "onClear": [Function],
                "placeholder": "Search for channels",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "buttonStyle": Object {
                  "backgroundColor": "transparent",
                },
                "containerStyle": Object {
                  "alignItems": "flex-start",
                  "flex": 0.1,
                  "justifyContent": "center",
                  "paddingBottom": 10,
                },
                "icon": <Themed.Icon
                  color="black"
                  name="add"
                  size={30}
                />,
                "onPress": [Function],
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Avenir-Black",
                  "fontSize": 24,
                }
              }
            >
              List
            </Text>,
            "style": Object {
              "marginBottom": 10,
              "marginLeft": 12,
              "marginTop": 5,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "List",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontFamily": "Avenir-Black",
                "fontSize": 24,
              },
            },
            "ref": null,
            "rendered": "List",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              undefined,
            ],
            "style": Object {
              "flex": 0.9,
              "flexDirection": "column",
              "marginHorizontal": 8,
            },
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": [Function],
        },
      ],
      "type": [MockFunction],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ScrollViewMock>
          <Component
            style={
              Object {
                "marginLeft": 10,
                "marginTop": 40,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Avenir-Black",
                  "fontSize": 24,
                }
              }
            >
              Search
            </Text>
          </Component>
          <Component
            style={
              Object {
                "flex": 0.13,
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <ForwardRef(Themed.SearchBar)
              clearIcon={false}
              containerStyle={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0,
                  "borderTopWidth": 0,
                  "flex": 0.9,
                  "justifyContent": "center",
                  "paddingBottom": 20,
                }
              }
              icon={
                Object {
                  "name": "search",
                  "type": "font-awesome",
                }
              }
              lightTheme={true}
              onChangeText={[Function]}
              onClear={[Function]}
              placeholder="Search for channels"
            />
            <ForwardRef(Themed.Button)
              buttonStyle={
                Object {
                  "backgroundColor": "transparent",
                }
              }
              containerStyle={
                Object {
                  "alignItems": "flex-start",
                  "flex": 0.1,
                  "justifyContent": "center",
                  "paddingBottom": 10,
                }
              }
              icon={
                <Themed.Icon
                  color="black"
                  name="add"
                  size={30}
                />
              }
              onPress={[Function]}
              title=""
            />
          </Component>
          <Component
            style={
              Object {
                "marginBottom": 10,
                "marginLeft": 12,
                "marginTop": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Avenir-Black",
                  "fontSize": 24,
                }
              }
            >
              List
            </Text>
          </Component>
          <Component
            style={
              Object {
                "flex": 0.9,
                "flexDirection": "column",
                "marginHorizontal": 8,
              }
            }
          />
        </ScrollViewMock>,
        "style": Object {
          "alignItems": "stretch",
          "backgroundColor": "#eee",
          "flex": 1,
          "paddingLeft": 8,
          "paddingRight": 8,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "marginLeft": 10,
                  "marginTop": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontFamily": "Avenir-Black",
                    "fontSize": 24,
                  }
                }
              >
                Search
              </Text>
            </Component>,
            <Component
              style={
                Object {
                  "flex": 0.13,
                  "flexDirection": "row",
                  "marginTop": 5,
                }
              }
            >
              <ForwardRef(Themed.SearchBar)
                clearIcon={false}
                containerStyle={
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": 0,
                    "borderTopWidth": 0,
                    "flex": 0.9,
                    "justifyContent": "center",
                    "paddingBottom": 20,
                  }
                }
                icon={
                  Object {
                    "name": "search",
                    "type": "font-awesome",
                  }
                }
                lightTheme={true}
                onChangeText={[Function]}
                onClear={[Function]}
                placeholder="Search for channels"
              />
              <ForwardRef(Themed.Button)
                buttonStyle={
                  Object {
                    "backgroundColor": "transparent",
                  }
                }
                containerStyle={
                  Object {
                    "alignItems": "flex-start",
                    "flex": 0.1,
                    "justifyContent": "center",
                    "paddingBottom": 10,
                  }
                }
                icon={
                  <Themed.Icon
                    color="black"
                    name="add"
                    size={30}
                  />
                }
                onPress={[Function]}
                title=""
              />
            </Component>,
            <Component
              style={
                Object {
                  "marginBottom": 10,
                  "marginLeft": 12,
                  "marginTop": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontFamily": "Avenir-Black",
                    "fontSize": 24,
                  }
                }
              >
                List
              </Text>
            </Component>,
            <Component
              style={
                Object {
                  "flex": 0.9,
                  "flexDirection": "column",
                  "marginHorizontal": 8,
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontFamily": "Avenir-Black",
                    "fontSize": 24,
                  }
                }
              >
                Search
              </Text>,
              "style": Object {
                "marginLeft": 10,
                "marginTop": 40,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Search",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontFamily": "Avenir-Black",
                  "fontSize": 24,
                },
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <ForwardRef(Themed.SearchBar)
                  clearIcon={false}
                  containerStyle={
                    Object {
                      "alignItems": "flex-start",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0,
                      "borderTopWidth": 0,
                      "flex": 0.9,
                      "justifyContent": "center",
                      "paddingBottom": 20,
                    }
                  }
                  icon={
                    Object {
                      "name": "search",
                      "type": "font-awesome",
                    }
                  }
                  lightTheme={true}
                  onChangeText={[Function]}
                  onClear={[Function]}
                  placeholder="Search for channels"
                />,
                <ForwardRef(Themed.Button)
                  buttonStyle={
                    Object {
                      "backgroundColor": "transparent",
                    }
                  }
                  containerStyle={
                    Object {
                      "alignItems": "flex-start",
                      "flex": 0.1,
                      "justifyContent": "center",
                      "paddingBottom": 10,
                    }
                  }
                  icon={
                    <Themed.Icon
                      color="black"
                      name="add"
                      size={30}
                    />
                  }
                  onPress={[Function]}
                  title=""
                />,
              ],
              "style": Object {
                "flex": 0.13,
                "flexDirection": "row",
                "marginTop": 5,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "clearIcon": false,
                  "containerStyle": Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": 0,
                    "borderTopWidth": 0,
                    "flex": 0.9,
                    "justifyContent": "center",
                    "paddingBottom": 20,
                  },
                  "icon": Object {
                    "name": "search",
                    "type": "font-awesome",
                  },
                  "lightTheme": true,
                  "onChangeText": [Function],
                  "onClear": [Function],
                  "placeholder": "Search for channels",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "buttonStyle": Object {
                    "backgroundColor": "transparent",
                  },
                  "containerStyle": Object {
                    "alignItems": "flex-start",
                    "flex": 0.1,
                    "justifyContent": "center",
                    "paddingBottom": 10,
                  },
                  "icon": <Themed.Icon
                    color="black"
                    name="add"
                    size={30}
                  />,
                  "onPress": [Function],
                  "title": "",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontFamily": "Avenir-Black",
                    "fontSize": 24,
                  }
                }
              >
                List
              </Text>,
              "style": Object {
                "marginBottom": 10,
                "marginLeft": 12,
                "marginTop": 5,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "List",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontFamily": "Avenir-Black",
                  "fontSize": 24,
                },
              },
              "ref": null,
              "rendered": "List",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                undefined,
              ],
              "style": Object {
                "flex": 0.9,
                "flexDirection": "column",
                "marginHorizontal": 8,
              },
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": [Function],
          },
        ],
        "type": [MockFunction],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
